pkg/controller/localvolumeset/localvolumeset_controller.go

```diff
@@ -100,54 +113,460 @@ func (r *ReconcileLocalVolumeSet) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	// Define a new Pod object
-	pod := newPodForCR(instance)
+	err = r.syncRBACPolicies(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	err = r.syncLocalVolumeSetDaemon(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	err = r.syncProvisionerConfigMap(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	err = r.syncProvisionerDaemonset(instance, r.getProvisionerConfigMapName(instance))
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
-	// Set LocalVolumeSet instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
+	localPVs := &corev1.PersistentVolumeList{}
+	err = r.client.List(context.TODO(), localPVs,
+		&client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set{
+				LocalVolumeSetNameLabel:      instance.GetName(),
+				LocalVolumeSetNamespaceLabel: instance.GetNamespace(),
+			}),
+		})
+	if err != nil {
 		return reconcile.Result{}, err
 	}
+	totalPVCount := int32(len(localPVs.Items))
+	instance.Status.TotalProvisionedDeviceCount = &totalPVCount
+	err = r.client.Status().Update(context.TODO(), instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
 
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
-		if err != nil {
-			return reconcile.Result{}, err
+func (r *ReconcileLocalVolumeSet) syncLocalVolumeSetDaemon(cr *localv1alpha1.LocalVolumeSet) error {
+	ds := newDiscoveryDaemonsetForCR(cr)
+	oldDs := &appsv1.DaemonSet{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: ds.Name, Namespace: ds.Namespace}, oldDs)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return r.client.Create(context.TODO(), ds)
 		}
-
-		// Pod created successfully - don't requeue
-		return reconcile.Result{}, nil
-	} else if err != nil {
-		return reconcile.Result{}, err
+		return err
+	}
+	if !reflect.DeepEqual(ds.Spec, oldDs.Spec) {
+		return r.client.Update(context.TODO(), oldDs)
```

Comment:
Looks like this was meant to modify `oldDS`, but it's not implemented?

Try https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrUpdate

Reply:
updated